File: D1-Morning/D1-S2-PHP/P5-UseRefOfAnotherObject/Salary.php
<?php


class Salary
{
    private $basic;
    private $conveyancePercent;
    private $medicalPercent;

    public function getBasic()
    {
        return $this->basic;
    }

    public function setBasic($basic): void
    {
        $this->basic = $basic;
    }

    public function getConveyancePercent()
    {
        return $this->conveyancePercent;
    }

    public function setConveyancePercent($conveyancePercent): void
    {
        $this->conveyancePercent = $conveyancePercent;
    }

    public function getMedicalPercent()
    {
        return $this->medicalPercent;
    }

    public function setMedicalPercent($medicalPercent): void
    {
        $this->medicalPercent = $medicalPercent;
    }



    public function getTotal(){
        return $this->basic + $this->getMedicalAmount() + $this->getConveyanceAmount();
    }

    public function getConveyanceAmount(){
        $amount = ($this->basic * $this->conveyancePercent)/100.0;
        return $amount;
    }
    public function getMedicalAmount(){
        $amount = ($this->basic * $this->medicalPercent)/100.0;
        return $amount;
    }
}

File: D1-Morning/C#/D1-S2/P5-UseRefOfAnotherObject/P5-UseRefOfAnotherObject/Salary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SolutionOfPractice1
{
    class Salary
    {
        public double Basic { set; get; }
        public double ConveyancePercent { set; get; }
        public double MedicalPercent { set; get; }

        public double GetTotal()
        {
            return Basic + GetMedicalAmount() + GetConveyanceAmount();
        }

        public double GetConveyanceAmount()
        {
            double amount = (Basic * ConveyancePercent) / 100.0;
            return amount;
        }

        public double GetMedicalAmount()
        {
            double amount = (Basic * MedicalPercent) / 100.0;
            return amount;        
        }


    }
}
